### PR TITLE
Frontend user state management with redux (plus a sign up page)

### PR DESCRIPTION
**Pull request by elimelt**
Wrote some basic redux actions and reducers to manage both the general global state of our app, as well as the user state in particular. Also integrated and styled (AKA stole @sriyabulusu's work) the CreateAccount component.

### CODE REVIEW

File: client/src/Components/CreateAccount/CreateAccount.js [hunk 1]
```diff
@@ -10,47 +15,64 @@ const Login = () => {
 
   const handlePasswordChange = event => setPassword(event.target.value)
 
-  const handleCreate = async event => {
+  const handleLogin = async event => {
```
Comment (elimelt) [inline client/src/Components/CreateAccount/CreateAccount.js]: Whoops!

File: client/src/Pages/SignUpPage.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import CreateAccount from '../Components/CreateAccount/CreateAccount'
+
+const SignUpPage = () => {
+  return (
+    <div className='page-container'>
+      <CreateAccount />
```
Comment (karhu1) [inline client/src/Pages/SignUpPage.js:6]: Will SignUpPage have more components than just this later on? Myabe slightly pointless to have it organized this way.

**Review by karhu1 — APPROVED** [review]
Looks great!

File: client/src/Pages/SignUpPage.js (hunk 2 above)
Comment (elimelt) [reply client/src/Pages/SignUpPage.js:6]: yes.

Comment (elimelt) [reply client/src/Pages/SignUpPage.js:6]: A longer answer is that CreateAccount is a single component that handles its internal styling and positioning, but nothing else outside of it. It has no bearing on the positioning within the larger page. The point of the page components (not just SignUpPage but every other page as well) is to handle the layout and positioning of things that are not contained within the component.

Keeping the CSS decoupled like this makes it much easier to style components without worrying about the contents of the rest of the page being pushed around or messed with.

This seems like a simpler, cleaner, and more maintainable solution than, say, adding a margin to the CreateAccount component to position it properly. This way, when we want to change the layout of the SignUpPage, we have a single consolidated CSS file to do so without having to delicately jump around all the CSS files that correspond to components that are only imported into the SignUpPage.

**Pull request merged**